APE17.rst

```diff
@@ -0,0 +1,701 @@
+| building C extensions               | and isolation in `PEP 517`_ and     |
++-------------------------------------+ `PEP 518`_                          |
+| Transpilation of Cython to C before |                                     |
+| sdist                               |                                     |
++-------------------------------------+                                     |
+| Package specific versions of        |                                     |
+| astropy-helpers provided by git     |                                     |
+| submodule                           |                                     |
++-------------------------------------+-------------------------------------+
+| ``--use-system-*`` options          | ``ASTROPY_USE_SYSTEM_*``            |
+|                                     | environment variables               |
++-------------------------------------+-------------------------------------+
+
+We propose that a new package called extension-helpers be created starting from
+astropy-helpers but with only functionality related to:
+
+* Collecting extension definitions from ``setup_package.py`` files
+* Auto-defining Cython extensions
+* Determining flags for OpenMP compilation
+
+This package would need to be declared as a build-time dependency in
+``pyproject.toml`` for packages that wish to use it.
+
+In addition to the changes described here, we also recommend moving all or as
+many as possible of the options for the ``setup()`` function in ``setup.py`` to
+the ``setup.cfg`` file as described in `the setuptools documentation
+<https://setuptools.readthedocs.io/en/latest/setuptools.html#configuring-setup-using-setup-cfg-files>`_.
+
+Timeline for changes
+--------------------
+
+All the changes described here can be made now. The recommendation in this APE
+is to first transition the core package and all coordinated packages to use this
+new infrastructure. In the process, the developer documentation for the core
+package as well as the package template must be updated to reflect the new
+infrastructure, and a migration guide must be written to help developers of
+affiliated and other packages using astropy-helpers change over to the new
+infrastructure.
+
+Once any issues have been ironed out in the core and coordinated packages, the
+migration guide should be advertised widely to all maintainers of packages
+using astropy-helpers. Throughout this process, astropy-helpers should continue
+to be supported, and once the migration guide is advertised the final date for
+support of astropy-helpers should be made clear. The recommendation in this
+APE is to support astropy-helpers until the end of the 4.0 LTS cycle at the
+latest (end of 2021), but this can be re-assessed depending on the ease of
+migration and the feasibility of maintaining astropy-helpers if there are
+extensive breaking changes in its dependencies.
+
+Impact for users
```

Comment:
```suggestion
Impact for users who are not developers
```

because on some level the "non-core developers" are really just users - this makes it less ambiguous

Reply:
Done